File: leetcode.com/0242_valid_anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sd = {}
        td = {}

        for c in s:
            sd.setdefault(c, 0)
            sd[c] += 1

        for c in t:
            td.setdefault(c, 0)
            td[c] += 1

        return sd == td

File: leetcode.com/0242_valid_anagram/solution.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }

    letters := map[rune]int{}
    
    for _, l := range s {
        letters[l]++
    }
    
    for _, l := range t {
        letters[l]--
    }
    
    for _, v := range letters {
        if v != 0 {
            return false
        }
    }
    
    return true
}
